[Unified PDF] Embedded PDFs are blank in snapshots

hortont424 opened the pull request:
#### 95b06f4446ccccd849b94e81168963f7cb7ceb2e
<pre>
[Unified PDF] Embedded PDFs are blank in snapshots
<a href="https://bugs.webkit.org/show_bug.cgi?id=278420">https://bugs.webkit.org/show_bug.cgi?id=278420</a>
<a href="https://rdar.apple.com/132092364">rdar://132092364</a>

Reviewed by Simon Fraser and Abrar Rahman Protyasha.

258399@main made us opt explicitly into local snapshot buffers for PDFPlugin
in the main frame, but not for subframes. We need local buffers in both cases,
since UnifiedPDFPlugin depends on having a platform context to paint into.

Check for PluginView (which is always PDFPlugin) instead of checking that the
frame itself is plugin-backed.

* Source/WebKit/WebProcess/WebPage/WebPage.cpp:
(WebKit::WebPage::snapshotAtSize):
(WebKit::snapshotImageOptions): Deleted.

* Tools/TestWebKitAPI/Tests/WebKitCocoa/UnifiedPDFTests.mm:
(TestWebKitAPI::configurationForWebViewTestingUnifiedPDF):
(TestWebKitAPI::TEST(UnifiedPDF, SnapshotsPaintPageContent)):
Add a test that confirms that the PDF content is painted.

Canonical link: <a href="https://commits.webkit.org/282551@main">https://commits.webkit.org/282551@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/5500ae8095c786a48e05d3f41b49d1b059713553

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/63489 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/42845 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/16086 "Built successfully") | [❌ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/67510 "") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/59/builds/14097 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/65609 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/50533 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/14377 "Built successfully") | [❌ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/5/builds/67510 "") | [  ~~🧪 wincairo-tests~~](https://ews-build.webkit.org/#/builders/60/builds/9752 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/66558 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/39778 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/54980 "Passed tests") | [❌ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/5/builds/67510 "") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/36462 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/64/builds/12348 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/12969 "Built successfully") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/13/builds/57973 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/12670 "Passed tests") | [❌ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/69206 "") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/87/builds/7436 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/62/builds/12252 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [❌ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/2/builds/69206 "") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/86/builds/7468 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/55061 "Passed tests") | [❌ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/2/builds/69206 "") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/14054 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/88/builds/6212 "Passed tests") | | | 
| | [  ~~🛠 tv~~](https://ews-build.webkit.org/#/builders/44/builds/38666 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
| | [  ~~🛠 tv-sim~~](https://ews-build.webkit.org/#/builders/46/builds/39745 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
| | [  ~~🛠 watch~~](https://ews-build.webkit.org/#/builders/43/builds/40857 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
| | [  ~~🛠 watch-sim~~](https://ews-build.webkit.org/#/builders/45/builds/39488 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
<!--EWS-Status-Bubble-End-->

aprotyas reviewed (approved):
This could likely be tested in the same way the `UnifiedPDF.KeyboardScrollingInSinglePageMode` API test is written

hortont424 committed:
[Unified PDF] Embedded PDFs are blank in snapshots
https://bugs.webkit.org/show_bug.cgi?id=278420
rdar://132092364

Reviewed by Simon Fraser and Abrar Rahman Protyasha.

258399@main made us opt explicitly into local snapshot buffers for PDFPlugin
in the main frame, but not for subframes. We need local buffers in both cases,
since UnifiedPDFPlugin depends on having a platform context to paint into.

Check for PluginView (which is always PDFPlugin) instead of checking that the
frame itself is plugin-backed.

* Source/WebKit/WebProcess/WebPage/WebPage.cpp:
(WebKit::WebPage::snapshotAtSize):
(WebKit::snapshotImageOptions): Deleted.

* Tools/TestWebKitAPI/Tests/WebKitCocoa/UnifiedPDFTests.mm:
(TestWebKitAPI::configurationForWebViewTestingUnifiedPDF):
(TestWebKitAPI::TEST(UnifiedPDF, SnapshotsPaintPageContent)):
Add a test that confirms that the PDF content is painted.

Canonical link: https://commits.webkit.org/282551@main
